src/gc/gc.cpp

```diff
@@ -36269,16 +36269,11 @@ CFinalize::ScanForFinalization (promote_func* pfn, int gen, BOOL mark_only_p,
 
                     assert (method_table(obj)->HasFinalizer());
 
-#ifndef FEATURE_REDHAWK
-                    if (method_table(obj) == pWeakReferenceMT || method_table(obj)->GetCanonicalMethodTable() == pWeakReferenceOfTCanonMT)
+                    if (GCToEEInterface::FinalizeIfWeakReference(obj))
```

Comment:
Right. I would call this EagerFinalized just to make it obvious that it returns a boolean value whether it was eagerly finalized; and if it was GC will do the management on the finalize queue.

---
In reply to: [109552827](https://github.com/dotnet/coreclr/pull/10676#discussion_r109552827) [](ancestors = 109552827)